astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -23,212 +168,638 @@ class NDArithmeticMixin(object):
     When subclassing, be sure to list the superclasses in the correct order
     so that the subclass sees NDData as the main superclass. See
     `~astropy.nddata.NDDataArray` for an example.
+
+    Notes
+    -----
+    This class only aims at covering the most common cases so there are certain
+    restrictions on the saved attributes::
+
+        - ``uncertainty`` : has to be something that has a `NDUncertainty`-like
```

Comment:
That was actually shortsighted of me, with the current implementation one only needed to pass the right ``propagate`` method for the uncertainty object. Probably I should switch back to the propagate-method parameter. But since there is no actual API what parameters should be passed to the ``propagate`` method makes it really hard to cover all cases.